pipeline_dp/accumulator.py

```diff
@@ -37,66 +38,92 @@ def add_accumulator(self, accumulator: 'Accumulator') -> 'Accumulator':
-    """Merges the accumulators of the CompoundAccumulators.
+    def add_accumulator(self, accumulator: 'CompoundAccumulator') -> \
+      'CompoundAccumulator':
+        """Merges the accumulators of the CompoundAccumulators.
 
     The expectation is that the internal accumulators are of the same type and
     are in the same order."""
 
-    if len(accumulator.accumulators) != len(self.accumulators):
-      raise ValueError(
-        "Accumulators in the input are not of the same size."
-        + f" Expected size = {len(self.accumulators)}"
-        + f" received size = {len(accumulator.accumulators)}.")
-
-    for pos, (base_accumulator_type, to_add_accumulator_type) in enumerate(
-      zip(self.accumulators, accumulator.accumulators)):
-      if type(base_accumulator_type) != type(to_add_accumulator_type):
-        raise TypeError("The type of the accumulators don't match at "
-                        f"index {pos}. {type(base_accumulator_type).__name__} "
-                        f"!= {type(to_add_accumulator_type).__name__}.")
-
-    for (base_accumulator, to_add_accumulator) in zip(self.accumulators,
-                                                      accumulator.accumulators):
-      base_accumulator.add_accumulator(to_add_accumulator)
-    return self
-
-  def compute_metrics(self):
-    """Computes and returns a list of metrics computed by internal
+        if len(accumulator.accumulators) != len(self.accumulators):
+            raise ValueError(
+                "Accumulators in the input are not of the same size." +
+                f" Expected size = {len(self.accumulators)}" +
+                f" received size = {len(accumulator.accumulators)}.")
+
+        for pos, (base_accumulator_type, to_add_accumulator_type) in enumerate(
+                zip(self.accumulators, accumulator.accumulators)):
+            if type(base_accumulator_type) != type(to_add_accumulator_type):
+                raise TypeError(
+                    "The type of the accumulators don't match at "
+                    f"index {pos}. {type(base_accumulator_type).__name__} "
+                    f"!= {type(to_add_accumulator_type).__name__}.")
+
+        for (base_accumulator,
+             to_add_accumulator) in zip(self.accumulators,
+                                        accumulator.accumulators):
+            base_accumulator.add_accumulator(to_add_accumulator)
+        return self
+
+    def compute_metrics(self):
+        """Computes and returns a list of metrics computed by internal
     accumulators."""
-    return [accumulator.compute_metrics() for accumulator in self.accumulators]
+        return [
+            accumulator.compute_metrics() for accumulator in self.accumulators
+        ]
+
+
+@dataclass
+class CountParams:
+    pass
+
+
+class CountAccumulator(Accumulator):
+
+    def __init__(self, params: CountParams, values):
+        self._count = len(values)
+
+    def add_value(self, value):
+        self._count += value
```

Comment:
Here it should be `+=1`, the idea that we're counting values (`value` will be element of collection)

Reply:
Done.